src/encoder.rs

```diff
@@ -351,6 +358,72 @@ impl<T: Pixel> FrameState<T> {
     }
   }
 
+  /// Computes the variance of 8x8 blocks factoring Contrast Sensitivity
+  /// Function in the tx_domain.
+  ///
+  /// # Panics
+  ///
+  /// Panics if the activity mask is not decimated by 8 on both axes.
+  pub fn compute_activity_mask(&mut self) {
```

Comment:
From what I can see it could be computed independently from the rest of the rest of encoding process, could you please make so that the function takes in the Plane + it own context data?

I would try to have it as a pre-process stage since it would be quite simple to run this in parallel over a group of frames before we run some inherently serial process down the road.

Reply:
Yes, I'll try it out.

Reply:
I have moved the computation of the activity mask to its own library. But I have not moved the call to a pre-processing stage.

Reply:
This looks OK. That said this single-line function is a bit superfluous.

Reply:
I agree. I'll remove the function.